diff --git a/gen-sdk.sh b/gen-sdk.sh
index 9f0bfaa..5ffe446 100755
--- a/gen-sdk.sh
+++ b/gen-sdk.sh
@@ -1,6 +1,6 @@
 #!/bin/bash
 
-set -e # Stop on error
+set -xe # Stop on error and print executed lines
 
 RM_If_Exists() {
     if [ -d $1 ] || [ -f $1 ]; then
@@ -19,7 +19,7 @@ Setup_SDK() {
 
     # Just in case
     set +e
-    sudo umount ./cache/${tc_target}/firmware*/mnt
+    # sudo umount ./cache/${tc_target}/firmware*/mnt
     set -e
     
     case $sdk_target in
@@ -106,7 +106,10 @@ Setup_SDK() {
 
           gunzip rootfs.img.gz
           mkdir -p mnt
-          sudo mount -o loop rootfs.img mnt
+          # sudo mount -o loop rootfs.img mnt
+          set +e # will error because it can't create two symlinks
+          7zz x rootfs.img -omnt/ -snld
+          set -e
       cd ../../..
     done
 
@@ -209,7 +212,7 @@ Setup_SDK() {
 
 
     echo "[*] Cleaning up"
-    sudo umount ./cache/${tc_target}/firmware*/mnt
+    # sudo umount ./cache/${tc_target}/firmware*/mnt
 
     echo "===================================================================================================="
     echo "[*] Kindle (unofficial) SDK Installed"
